File: Maximum Depth of Binary Tree.cpp
// Leetcode Problem 104: Maximum Depth of Binary Tree
// C++ CODE
class Solution {
public:
    int maxDepth(TreeNode* root) {

        // Empty tree
        if (root == nullptr) {
            return 0;
        }

        // Find depth of left and right subtrees
        int leftDepth = maxDepth(root->left);
        int rightDepth = maxDepth(root->right);

        // Current node adds 1
        return 1 + max(leftDepth, rightDepth);
    }
};

File: Maximum Depth of Binary Tree.py
# Leetcode Problem 104: Maximum Depth of Binary Tree
# PYTHON CODE
class Solution:
    def maxDepth(self, root):
        
        # Empty tree
        if root is None:
            return 0
        
        # Find depth of left and right subtrees
        leftDepth = self.maxDepth(root.left)
        rightDepth = self.maxDepth(root.right)
        
        # Current node adds 1 to the maximum depth
        return 1 + max(leftDepth, rightDepth)